libs/librepcb/rust-core/ffi.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * LibrePCB - Professional EDA for everyone!
+ * Copyright (C) 2013 LibrePCB Developers, see AUTHORS.md for contributors.
+ * https://librepcb.org/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*******************************************************************************
+ *  Includes
+ ******************************************************************************/
+#include "ffi.h"
+
+/*******************************************************************************
+ *  Namespace
+ ******************************************************************************/
+namespace librepcb {
+namespace rs {
+
+/*******************************************************************************
+ *  Functions
+ ******************************************************************************/
+
+extern "C" std::size_t ffi_qbytearray_len(const QByteArray* obj) {
+  return static_cast<std::size_t>(obj->size());
+}
+
+extern "C" const uint8_t* ffi_qbytearray_data(const QByteArray* obj) {
+  return reinterpret_cast<const uint8_t*>(obj->data());
```

Comment:
```suggestion
  return reinterpret_cast<const uint8_t*>(obj->constData());
```

There is an explicit `constData`, but I'm not sure if it has any benefits.

Reply:
Shouldn't have benefits here. AFAIK it's to avoid detaching the COW type `QByteArray` when called on a non-const object, but our `obj` is `const QByteArray*` so it won't detach anyway.